File: Python/7py 1/accum.py
"""
|--------------------------------------------------------------------------
| Mumbling
|--------------------------------------------------------------------------
|
"""

def accum(s):
   arr = []
   split = list(s.lower())
   for i in range(len(split)):
      res = ''
      for j in range(i+1):
         if j == 0:
            res += split[i].upper()
         else:
            res += split[i]
      arr.append(res)
   return '-'.join(arr)


def accum(s):
   return '-'.join(c.upper() + c.lower() * i for i, c in enumerate(s))


def accum(s):
   output = ""
   for i in range(len(s)):
      output+=(s[i]*(i+1))+"-"
   return output.title()[:-1]


def accum(s):
   return '-'.join((a * i).title() for i, a in enumerate(s, 1))


def accum(s):
   return "-".join([((j*(i+1)).capitalize()) for i,j in enumerate(s)])


print(accum("ZpglnRxqenU"), "Z-Pp-Ggg-Llll-Nnnnn-Rrrrrr-Xxxxxxx-Qqqqqqqq-Eeeeeeeee-Nnnnnnnnnn-Uuuuuuuuuuu")
print(accum("NyffsGeyylB"), "N-Yy-Fff-Ffff-Sssss-Gggggg-Eeeeeee-Yyyyyyyy-Yyyyyyyyy-Llllllllll-Bbbbbbbbbbb")

File: PHP/accum.php
<?php
/*
|--------------------------------------------------------------------------
| Mumbling
|--------------------------------------------------------------------------
|
*/

function accum($s) {
   $arr = [];
   $split = str_split(strtolower($s));
   for ($i=0; $i < count($split); $i++) { 
      $res = '';
      for ($j=0; $j <= $i; $j++) {
         $res .= $j == 0 ? strtoupper($split[$i]) : $split[$i];
      }
      array_push($arr, $res);
   }
   return join('-', $arr);
}


function accum($s) {
   foreach (str_split($s) as $position => $char) {
      $words[] = ucwords(str_repeat(strtolower($char), $position + 1));
   }
   return implode('-', $words);
}


function accum($s) {
   $charray = str_split($s);
   array_walk($charray, function(&$char, $index) {
      $char = strtoupper($char) . str_repeat(strtolower($char), $index);
   });
   return implode('-', $charray);
}


function accum($s) {
   $result = [];
   for ($i = 0; $i < strlen($s); $i++) {
      array_push($result, ucfirst(strtolower(str_repeat($s[$i], ($i + 1)))));
   }
   return implode("-",$result);
}


print_r(accum("ZpglnRxqenU"). "Z-Pp-Ggg-Llll-Nnnnn-Rrrrrr-Xxxxxxx-Qqqqqqqq-Eeeeeeeee-Nnnnnnnnnn-Uuuuuuuuuuu
");
print_r(accum("NyffsGeyylB"). "N-Yy-Fff-Ffff-Sssss-Gggggg-Eeeeeee-Yyyyyyyy-Yyyyyyyyy-Llllllllll-Bbbbbbbbbbb
");